nodejs/eks/nodegroup.ts

```diff
@@ -229,16 +234,29 @@ export function createNodeGroup(name: string, args: NodeGroupOptions, parent: pu
     const awsRegion = pulumi.output(aws.getRegion({}, { parent: parent }));
     const userDataArg = args.nodeUserData || pulumi.output("");
 
-    let bootstrapExtraArgs = "";
+    const kubeletExtraArgs: Array<string> = [];
     if (args.labels) {
         const parts = [];
         for (const key of Object.keys(args.labels)) {
             parts.push(key + "=" + args.labels[key]);
         }
         if (parts.length > 0) {
-            bootstrapExtraArgs = " --kubelet-extra-args --node-labels=" + parts.join();
+            kubeletExtraArgs.push("--node-labels=" + parts.join(","));
         }
     }
+    if (args.taints) {
+        const parts = [];
+        for (const key of Object.keys(args.taints)) {
+            parts.push(key + "=" + args.taints[key]);
+        }
+        if (parts.length > 0) {
+            kubeletExtraArgs.push("--register-with-taints=" + parts.join(","));
+        }
+    }
+    let bootstrapExtraArgs = "";
+    if (kubeletExtraArgs.length > 0) {
+        bootstrapExtraArgs = ` --kubelet-extra-args '${kubeletExtraArgs.join(" ")}'`;
```

Comment:
One slight potential issue - the addition of `'` around the args will cause any existing node groups to see a change and force all nodes to get replaced as part of an upgrade to the latest version of the `@pulumi/eks` package.

We could avoid this by only adding the surrounding `'` when the length is greater than 1.